packages/protocol/migrations_sol/create_and_migrate_anvil_devchain.sh

```diff
@@ -1,73 +1,69 @@
 #!/usr/bin/env bash
 set -euo pipefail
 
-# Compile everything
-forge build
+# Keeping track of start time to measure how long it takes to run the script entirely
+START_TIME=$SECONDS
 
 export ANVIL_PORT=8546
 
 # TODO make this configurable
 FROM_ACCOUNT_NO_ZERO="f39Fd6e51aad88F6F4ce6aB8827279cffFb92266" # This is Anvil's default account (1)
-FROM_ACCOUNT="0x$FROM_ACCOUNT_NO_ZERO"
+export FROM_ACCOUNT="0x$FROM_ACCOUNT_NO_ZERO"
+
+# Create temporary directory
 TEMP_FOLDER="$PWD/.tmp"
+mkdir -p $TEMP_FOLDER
 
+# Start a local anvil instance
 source $PWD/migrations_sol/start_anvil.sh
 
-source $PWD/migrations_sol/deploy_precompiles.sh
-
-
-echo "Setting Registry Proxy"
-REGISTRY_ADDRESS="0x000000000000000000000000000000000000ce10"
-PROXY_BYTECODE=`cat ./out/Proxy.sol/Proxy.json | jq -r '.deployedBytecode.object'`
-cast rpc anvil_setCode --rpc-url http://127.0.0.1:$ANVIL_PORT $REGISTRY_ADDRESS $PROXY_BYTECODE
-REGISTRY_OWNER_ADDRESS=$FROM_ACCOUNT_NO_ZERO
-
-echo "Setting Registry owner"
-# Sets the storage of the registry so that it has an owner we control
-# pasition is bytes32(uint256(keccak256("eip1967.proxy.admin")) - 1);
-cast rpc anvil_setStorageAt --rpc-url http://127.0.0.1:$ANVIL_PORT $REGISTRY_ADDRESS 0xb53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d6103 "0x000000000000000000000000$REGISTRY_OWNER_ADDRESS"
-
-
-echo "Deploying libraries"
-LIBRARIES_PATH=("contracts/common/linkedlists/AddressSortedLinkedListWithMedian.sol:AddressSortedLinkedListWithMedian"
-                "contracts/common/Signatures.sol:Signatures"
-                "contracts/common/linkedlists/AddressLinkedList.sol:AddressLinkedList"
-                "contracts/common/linkedlists/AddressSortedLinkedList.sol:AddressSortedLinkedList"
-                "contracts/common/linkedlists/IntegerSortedLinkedList.sol:IntegerSortedLinkedList"
-                "contracts/governance/Proposals.sol:Proposals"
-)
-
-LIBRARIES=""
+# Deploy libraries to the anvil instance
+source $PWD/migrations_sol/deploy_libraries.sh
 
-for library in "${LIBRARIES_PATH[@]}"; do
-    library_name="${library#*:}" 
-    echo "Deploying library: $library_name"
-    create_library_out=`forge create $library --from $FROM_ACCOUNT --rpc-url http://127.0.0.1:$ANVIL_PORT --unlocked --json`
-    library_address=`echo $create_library_out | jq -r '.deployedTo'`
-    
-    LIBRARIES="$LIBRARIES --libraries $library:$library_address"
-done
-
-echo "Library flags are: $LIBRARIES"
+# Backing up library flags
+# TODO(Arthur): Check if this is strictly necessary. 
```

Comment:
Nit: I'll briefly look into this and address or remove the `TODO` comment. 
Any changes will make CI run again, but shouldn't significantly change the logic of the script.

Reply:
Removed comment in https://github.com/celo-org/celo-monorepo/pull/11003/commits/5917a6a192c15f4b6be442ea82de457748ff15a5